silkworm/sentry/rlpx/peer.cpp

```diff
@@ -104,7 +104,11 @@ Task<void> Peer::handle() {
     bool is_handshake_completed = false;
     [[maybe_unused]] auto _ = gsl::finally([this, &is_handshake_completed] {
         if (!is_handshake_completed) {
-            this->handshake_promise_.set_value(false);
+            try {
+                this->handshake_promise_.set_value(false);
+            } catch (const std::exception& e) {
```

Comment:
what kind of exception is thrown and why?
is it possible to avoid a "catch all" and "ignore" here?

Reply:
It's a `std::runtime_error` we raise in `AwaitableFuture::set` when `AsyncChannel::try_send` returns false (i.e. not sent) and the channel is still open. We definitely can investigate this issue more and probably have a better fix, but I propose to do this in next PR on the subject.

Maybe we can to keep #1896 open and then you can put it on your todo list, if you agree